crates/astria-conductor/src/conductor.rs

```diff
@@ -45,20 +51,39 @@ use crate::{
         self,
         ClientProvider,
     },
+    config::CommitLevel,
     data_availability::{
         self,
         CelestiaReaderConfig,
     },
-    executor::Executor,
+    // executor,
+    executor::{
+        Executor,
+        // ExecutorCommand::GetExecutableBlockHeight,
+        ExecutorCommitmentState,
+    },
     sequencer,
     Config,
 };
 
 pub struct Conductor {
-    /// The data availability reader that is spawned after sync is completed.
-    /// Constructed if constructed if `disable_finalization = false`.
+    /// The sequencer reader that is spawned after either after DA sync is completed,
+    /// or when running in "SoftOnly".
+    sequencer_reader: Option<sequencer::Reader>,
+
+    /// The data availability syncer that is spawned for DA sync, or after
+    /// sequencer sync has been completed.
+    data_availability_syncer: Option<data_availability::Reader>,
```

Comment:
I am not sure I understand the design. Why is there a separate data availability syncer and reader? I see that conductor has a `sync_done` field, but it seems unnecessary given the 2 separate actors here?